lib/sisimai/reason/rejected.rb

```diff
@@ -78,23 +82,39 @@ def match(argv1)
         def true(argvs)
           return nil unless argvs
           return nil unless argvs.is_a? Sisimai::Data
-          return nil unless argvs.deliverystatus.size > 0
-          return true if argvs.reason == Sisimai::Reason::Rejected.text
 
           require 'sisimai/smtp/status'
           statuscode = argvs.deliverystatus || ''
-          diagnostic = argvs.diagnosticcode || ''
           reasontext = Sisimai::Reason::Rejected.text
+
+          return true if argvs.reason == reasontext
+
+          tempreason = Sisimai::SMTP::Status.name(statuscode)
+          tempreason = 'undefined' if tempreason.empty?
+          diagnostic = argvs.diagnosticcode || ''
           v = false
 
-          if Sisimai::SMTP::Status.name(statuscode) == reasontext
-            # Delivery status code points C<rejected>.
+          if tempreason == reasontext
+            # Delivery status code points "rejected".
             v = true
           else
             # Check the value of Diagnosic-Code: header with patterns
             if argvs.smtpcommand == 'MAIL'
-              # Matched with a pattern in this class
+              # The session was rejected at 'MAIL FROM' command
               v = true if Sisimai::Reason::Rejected.match(diagnostic)
+
+            elsif argvs.smtpcommand == 'DATA'
+              # The session was rejected at 'DATA' command
+              if tempreason != 'userunknown'
+                # Except "userunknown"
+                v = true if Sisimai::Reason::Rejected.match(diagnostic)
+              end
+            else
+              if tempreason == 'undefined' || tempreason == 'onhold'
```

Comment:
__[RuboCop]__

Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.


<!-- Comment by [SideCI](https://sideci.com) -->